src/util/endpoints.js

```diff
@@ -124,6 +124,33 @@ exports.getEndpoint = async (switchUrl, fsp, endpointType, options = {}) => {
   }
 }
 
+/**
+ * @function getEndpointAndRender
+ *
+ * @description It returns the rendered endpoint for a given fsp and type from the cache if the cache is still valid, otherwise it will refresh the cache and return the value
+ *
+ * @param {string} switchUrl the endpoint for the switch
+ * @param {string} fsp - the id of the fsp
+ * @param {string} endpointType - the type of the endpoint
+ * @param {string} path - callback endpoint path
+ * @param {object} options - contains the options for the mustache template function
+ *
+ * @returns {string} - Returns the rendered endpoint, throws error if failure occurs
+ */
+exports.getEndpointAndRender = async (switchUrl, fsp, endpointType, path = '', options) => {
+  switchEndpoint = switchUrl
+  Logger.isInfoEnabled && Logger.info(`participantEndpointCache::getEndpointAndRender::endpointType - ${endpointType}`)
+  try {
+    const endpoints = await policy.get(fsp)
+    const endpoint = new Map(endpoints).get(endpointType)
+    const renderedEndpoint = (endpoint === undefined) ? endpoint : endpoint + path
```

Comment:
```suggestion
    const renderedEndpoint = (endpoint || '') + path
```

Reply:
I wanted this to handle an undefined `endpoint` exactly like `getEndpoint` where it passes `undefined` to `Mustache.render` where it subsequently throws an rendering error.

(endpoint || '') + path can end up with a malformed string like "/parties" instead of "my-host:port/parties"